server/routes/getAvailability.js

```diff
@@ -0,0 +1,144 @@
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+
+//gets all the teachers from the Teacher table
+async function getAllTeachers() {
+
+    const allTeachers = await prisma.Teacher.findMany();
+    return allTeachers;
+
+}
+//gets all the schedule from Schedule table
+async function getAllSchedule(){
+    const allSchedule = await prisma.Schedule.findMany();
+    return allSchedule;
+}
+//gets all the schedule from the ScheduledClass table
+async function getAllScheduledClass() {
+    const allSchedule = await prisma.ScheduledClass.findMany();
+    return allSchedule;
+}
+
+//gets all the absences from the absence table
+async function getAbsences(date) {
+    const absences = await prisma.Absence.findMany( {
+        where: {
+            day: date
+        },
+        select: {
+            id: true,
+            teacherId: true,
+            day: true,
+            period: true
+        }
+    });
+
+    return absences;
+}
+
+//returns single schedule id with all the periods associated with it
+async function getScheduleIdAndPeriods () {
```

Comment:
This is an okay way to do this, but you can do it more efficiently with the right prisma call:
```
  const schedules = await prisma.schedule.findMany({
    select: {
      teacher: true,
      classes: true,
      id: true
    }
  });
 const scheduleIdAndPeriods = schedules.map(x => ({scheduleId: x.id, periods: x.classes.map(y => y.period)}));
``` 
to be clear i don't think anything has to change here, just helpful to know

Reply:
Then you can also just include the teacherId instead of having the `getFreePeriodsWithTeacherId` function that parses the lists again